### PR TITLE
clickable: init at 7.4.0

### PR DESCRIPTION
**Pull request by ilyakooo0**
###### Description of changes

Adds the clickable tool used to build and run Ubuntu Touch apps.

Homepage: https://clickable-ut.dev/en/latest/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/clickable/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  propagatedBuildInputs = [
+    cookiecutter
+    requests
+    pyyaml
+    jsonschema
+    argcomplete
+  ];
+
+  # Logging test try to write to ~/.clickable/clickable.log
+  doCheck = false;
```
Comment (legendofmiracles) [inline pkgs/development/tools/clickable/default.nix:30]:
```suggestion
  preCheck = ''
    # Tests use the home directory
    export HOME="$(mktemp -d)"
  ''
```
This should allow the program to access a home directory

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

**Comment by onny** [comment]
Would close my PR in favor of this in case it gets merged :) https://github.com/NixOS/nixpkgs/pull/177731

File: pkgs/development/tools/clickable/default.nix (hunk 1 above)
Comment (ilyakooo0) [reply pkgs/development/tools/clickable/default.nix]: when I set that, the logging tests fail in another way:

<details>
<summary>Log
</summary>
```
Traceback (most recent call last):
  File "/private/tmp/nix-build-python3.9-clickable-7.4.0.drv-0/source/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 29, in <module>
    setup(
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 148, in setup
    return run_commands(dist)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
    dist.run_commands()
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
    self.run_command(cmd)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/dist.py", line 1214, in run_command
    super().run_command(command)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/test.py", line 224, in run
    self.run_tests()
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/test.py", line 227, in run_tests
    test = unittest.main(
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 124, in parseArgs
    self._do_discovery(argv[2:])
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 244, in _do_discovery
    self.createTests(from_discovery=True, Loader=Loader)
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/main.py", line 154, in createTests
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 349, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 405, in _find_tests
    tests, should_recurse = self._find_test_path(
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 483, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/nix/store/sw1nv9pjardbng91xg1xfwgyd1bw98mp-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/test.py", line 57, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/4yl4jm5f3av4rr21ikpmh4ad8z6f7ic5-python3-3.9.12/lib/python3.9/unittest/loader.py", line 214, in loadTestsFromName
    raise TypeError("don't know how to make test from: %s" % obj)
TypeError: don't know how to make test from: <Logger clickable (DEBUG)>
```
</details>

I am a bit time constrained at the moment, and don't have the ability to dig deeper (:

File: pkgs/development/tools/clickable/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    requests
+    pyyaml
+    jsonschema
+    argcomplete
+  ];
+
+  preCheck = ''
+    # Tests use the home directory
+    export HOME="$(mktemp -d)"
+  ''
```
Comment (legendofmiracles) [inline pkgs/development/tools/clickable/default.nix:33]:
```suggestion
  # tests fail with: TypeError: don't know how to make test from
  doCheck = false;
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -12345,6 +12345,8 @@ with pkgs;
 
   clean = callPackage ../development/compilers/clean { };
 
+  clickable = python3Packages.callPackage ../development/tools/clickable {};
```
Comment (legendofmiracles) [inline pkgs/top-level/all-packages.nix:12348]:
```suggestion
  clickable = python3Packages.callPackage ../development/tools/clickable { };
```

**Review by legendofmiracles — COMMENTED** [review]
If it doesn't work we can revert it again I guess

File: pkgs/development/tools/clickable/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, fetchFromGitLab
+, buildPythonPackage
+, cookiecutter
+, requests
+, pyyaml
+, jsonschema
+, argcomplete
```
Comment (onny) [inline pkgs/development/tools/clickable/default.nix:8]:
```suggestion
, argcomplete
, pytestCheckHook
```

File: pkgs/development/tools/clickable/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    cookiecutter
+    requests
+    pyyaml
+    jsonschema
+    argcomplete
+  ];
+
+  # tests fail with: TypeError: don't know how to make test from
+  doCheck = false;
+
```
Comment (onny) [inline pkgs/development/tools/clickable/default.nix:32]:
```suggestion
checkInputs = [ pytestCheckHook ];

disabledTests = [
  # Test require network connection
  "test_cpp_plugin"
  "test_html"
  "test_python"
  "test_qml_only"
  "test_rust"
];
```

**Review by onny — CHANGES_REQUESTED** [review]
After adding the suggestion from @legendofmiracles and disabling failing tests, the package builds fine for me :)

File: pkgs/development/tools/clickable/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    cookiecutter
+    requests
+    pyyaml
+    jsonschema
+    argcomplete
+  ];
+
+  # tests fail with: TypeError: don't know how to make test from
+  doCheck = false;
+checkInputs = [ pytestCheckHook ];
```
Comment (onny) [inline pkgs/development/tools/clickable/default.nix:33]: Please fix the indention here

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (onny) [inline pkgs/top-level/all-packages.nix:12348]: Please fix formatting issue here

File: pkgs/development/tools/clickable/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  propagatedBuildInputs = [
+    cookiecutter
+    requests
+    pyyaml
+    jsonschema
+    argcomplete
+  ];
+
+  # tests fail with: TypeError: don't know how to make test from
+  doCheck = false;
```
Comment (onny) [inline pkgs/development/tools/clickable/default.nix:32]: Tests worked for me, so we can remove this statement

**Review by onny — CHANGES_REQUESTED** [review]

**Pull request merged**